### PR TITLE
Add build badges and update test coverage config #482

### LINKED ISSUE #482: Add build badges and update test coverage config
What should be done:
1. Add CI build badge at the top of the https://github.com/ubiquity/ubiquity-dollar/blob/development/README.md with the current CI build status (success or fail)
2. Add test coverage badge with % of lines covered by tests at the top of the https://github.com/ubiquity/ubiquity-dollar/blob/development/README.md
3. Remove the following files from the test coverage report:

- `src/dollar/libs/*`
- `src/dollar/mocks/*`
- `src/dollar/utils/*`
- `src/manager/libraries/*`
- `src/ubiquistick/MockUBQmanager.sol`
- `test/*`

Build badge docs: https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge

### PR DESCRIPTION
**Pull request by QuintusTheFifth**
Resolves #482

### CODE REVIEW

**Comment by 0x4007** [comment]
@rndquu the readme is looking as expected https://github.com/QuintusTheFifth/ubiquity-dollar#readme could you confirm this is implemented as you envisioned?

@QuintusTheFifth unfortunately it appears that your first commit was not following conventional commits. Can you fix that?

File: .github/workflows/coverage-check.yml [hunk 1]
```diff
@@ -49,7 +49,15 @@ jobs:
               --rc lcov_branch_coverage=1 \
               --remove merged-lcov.info \
               --output-file filtered-lcov.info \
-              "*node_modules*" "*test*" "*mock*" "*scripts*"
+                  "*node_modules*" \
+                  "*test*" \
+                  "*mock*" \
+                  "*scripts*" \
+                  "src/dollar/libraries/*" \
```
Comment (rndquu) [inline .github/workflows/coverage-check.yml]: In the original issue the `src/dollar/libs/*` path is mentioned to be removed from coverage (where we used to store mocked libs). The `src/dollar/libraries/*` path actually contains facet libs which should be tested. Could you remove this `src/dollar/libraries/*` path from coverage filters?

File: packages/contracts/coverage.sh [hunk 2]
```diff
@@ -20,8 +20,16 @@ lcov \
...
     --remove merged-lcov.info \
     --output-file filtered-lcov.info \
-    "*node_modules*" "*test*" "*mock*" "*scripts*"
-
+    --ignore-errors unused \
+    "*node_modules*" \
+    "*test*" \
+    "*mock*" \
+    "*scripts*" \
+    "src/dollar/libraries/*" \
```
Comment (rndquu) [inline packages/contracts/coverage.sh]: Again, pls remove the `src/dollar/libraries/*` path as it should be processed by the coverage script

**Review by rndquu — CHANGES_REQUESTED** [review]
@QuintusTheFifth awesome

almost ready to be merged, pls fix some minor issues

**Comment by rndquu** [comment]
@QuintusTheFifth we need to set the `COVERALLS_REPO_TOKEN` secret in our own environment, right?

File: .github/workflows/coverage-check.yml (hunk 1 above)
Comment (QuintusTheFifth) [reply .github/workflows/coverage-check.yml]: ok, updated it

File: packages/contracts/coverage.sh (hunk 2 above)
Comment (QuintusTheFifth) [reply packages/contracts/coverage.sh]: ok, changed as well

**Comment by QuintusTheFifth** [comment]
> @QuintusTheFifth we need to set the `COVERALLS_REPO_TOKEN` secret in our own environment, right? 

exactly, and the link for the coverage status should be changed to the ubiquity repo @rndquu

**Comment by QuintusTheFifth** [comment]
> @QuintusTheFifth unfortunately it appears that your first commit was not following conventional commits. Can you fix that?

yes, fixed it just now @pavlovcik

**Comment by 0x4007** [comment]
Seems like everything is addressed here. @rndquu when you approve, please merge and bypass zgo's review.

**Comment by rndquu** [comment]
@pavlovcik could you:
1. Authorize at https://coveralls.io/

**Comment by rndquu** [comment]
@pavlovcik could you:
1. Authorize at https://coveralls.io/
2. Enable coveralls for the https://github.com/ubiquity/ubiquity-dollar repo
3. Share coveralls repo token for that repo (or set it yourself as a `COVERALLS_REPO_TOKEN` repository secret for the https://github.com/ubiquity/ubiquity-dollar repo)

**Review by rndquu — APPROVED** [review]

**Comment by pavlovcik** [comment]
> Share coveralls repo token for that repo (or set it yourself as a COVERALLS_REPO_TOKEN repository secret for the https://github.com/ubiquity/ubiquity-dollar repo)

I am trying to figure this out now.

**Comment by 0x4007** [comment]
All set!

**Comment by pavlovcik** [comment]
It says "coverage unknown" on the README. This seems to not be working.

**Pull request merged**